terraform/README.adoc

```diff
@@ -82,45 +99,92 @@ overlay
+  users.extraGroups.lxd.members = [ "your-user" ];
+  users.extraGroups.lxc.members = [ "your-user" ];
+
+  # Following line needed only if you choose LVM backend for LXC
+  # Following line is a workaround for the issue of lvm tools not being
+  # in the PATH of LXD (https://github.com/NixOS/nixpkgs/issues/31117)
+  systemd.services.lxd.path = with pkgs; [ lvm2 thin-provisioning-tools e2fsprogs ];
+
+  # Needed for kube-proxy pod that crashes if the hashsize is not big enough.
+  # It can't be modified from inside the container even if sys is mounted rw.
+  boot.extraModprobeConfig = ''
+    options nf_conntrack hashsize=393216
+  '';
 ----
- virtualisation.lxd = {
-    enable = true;
-    zfsSupport = false;
-  };
 
- users.users.gila = {
-    isNormalUser = true;
-    extraGroups = [ "wheel" ... "lxd" "lxc" ];
-  };
+After that run `lxd init` and configure it to suit k8s cluster and your
+needs. In particular do not use btrfs or ZFS as a storage pool. Docker's
+AUFS storage driver does not work with them out of the box.
+Example of a seed file generated by lxd init for the case of LVM
+backend with storage pool on `/dev/sdb1` and network `192.168.2.1/24`
+used for the lxc containers:
 
-nixpkgs.config.packageOverrides = super: let self = super.pkgs; in {
-    lxc = super.lxc.overrideAttrs (oldAttrs: rec {
-      patches = oldAttrs.patches ++ [
-        (self.fetchpatch {
-          url = "https://github.com/lxc/lxc/commit/b31d62b847a3ee013613795094cce4acc12345ef.patch";
-          sha256 = "1jpskr58ih56dakp3hg2yhxgvmn5qidi1vzxw0nak9afbx1yy9d4";
-        })
-      ];
-    });
-  };
+[source,yaml]
+----
+config: {}
+cluster: null
+networks:
+- config:
+    ipv4.address: 192.168.2.1/24
+    ipv4.nat: "false"
+    ipv6.address: auto
+  description: ""
+  managed: false
+  name: lxdbr0
+  type: ""
+storage_pools:
+- config:
+    source: /dev/sdb1
+  description: ""
+  name: lxd-pool
+  driver: lvm
+profiles:
+- config: {}
+  description: ""
+  devices:
+    eth0:
+      name: eth0
+      nictype: bridged
+      parent: lxdbr0
+      type: nic
+    root:
+      path: /
+      pool: lxd-pool
+      type: disk
+  name: default
 ----
 
-After that run `lxd init` and configure it to best suit your needs.
+It is *important* to test that LXD works before you move to terraform
+apply step. Create a container and test that it can reach the internet.
 
-WARNING: do not use btrfs or ZFS as a storage pool.
+[source,bash]
+----
+lxc launch ubuntu:18.04 first
+lxc exec first -- /bin/bash -c 'wget http://google.com/'
+----
 
-Now comes a somewhat tricky part. We need to install the terraform lxd provider,
-but it does not exist in nixpkgs. I'll create a PR to NixOS shortly.
+=== Setting up LXD with terraform on other distros than NixOS
 
-On other distributions, manually install the lxd provider from
+Read the above section and ignore NixOs specific configuration and
+workarounds. When it comes to installing terraform with LXD provider,
```

Comment:
Could the earlier steps be copied and modified rather than referring the reader those steps but requiring them to modify on-the-fly?  Then the thinking needed can be done once, rather than every time.

Reply:
right. I'm reworking this section and splitting it into three sections: common instructions, nixos specific, other distros specific.